src/lib/containers/UserCard.tsx

```diff
@@ -66,74 +67,37 @@ export const UserCard: React.FunctionComponent<UserCardProps> = (
   // We fetch the image right away in case it is an expiring presigned URL
   const imageURL = usePreFetchResource(preSignedURL)
 
+  const { data: profileAndImage } = useGetUserProfileWithProfilePic(
+    principalId!,
+    {
+      enabled: !!principalId,
+    },
+  )
+
+  useEffect(() => {
+    if (profileAndImage) {
+      if (!initialProfile) {
+        setUserProfile(profileAndImage.userProfile)
+      }
+      if (!initialPreSignedURL) {
+        setPresignedUrl(profileAndImage.preSignedURL)
+      }
+    }
+  }, [profileAndImage, initialPreSignedURL, initialProfile])
+
   useEffect(() => {
     if (userProfile) {
       setIsLoading(false)
     } else if (alias) {
       // Before we can get the profile, we must get the principal ID using the alias
       getPrincipalAliasRequest(accessToken, alias, 'USER_NAME').then(
-        (aliasData: any) => {
-          setPrincipalId(aliasData.principalId)
+        aliasData => {
+          setPrincipalId(aliasData.principalId.toString())
         },
       )
     }
   }, [userProfile, alias, accessToken])
 
-  useEffect(() => {
-    const getProfilePicture = (
-      profileOwnerId: string,
-      fileHandleId?: string,
-    ) => {
-      if (fileHandleId) {
-        const fha = {
-          associateObjectId: profileOwnerId,
-          associateObjectType: 'UserProfileAttachment',
-          fileHandleId: fileHandleId,
-        }
-        const request: any = {
-          includeFileHandles: false,
-          includePreSignedURLs: true,
-          includePreviewPreSignedURLs: false,
-          requestedFiles: [fha],
-        }
-        SynapseClient.getFiles(request, accessToken)
-          .then(fileHandleList => {
-            setPresignedUrl(fileHandleList.requestedFiles[0].preSignedURL!)
-            setIsLoading(false)
-          })
-          .catch(err => {
-            console.warn('failed to get user profile picture ', err)
-          })
-      } else {
-        setIsLoading(false)
-      }
-    }
-
-    if (!userProfile && principalId) {
-      const cachedProfileString = sessionStorage.getItem(
```

Comment:
In synapse.org, will any cache be referenced for the UserCard data?  
I think this was using a session storage cache established on synapse.org.

Reply:
Aha: https://github.com/Sage-Bionetworks/SynapseWebClient/blob/f70b3a781694c2bd5002f0722c66bf07b3029e6b/src/main/java/org/sagebionetworks/web/shared/WebConstants.java#L244